### ISSUE TITLE
FPRTruncatedURLString crash issue

### ISSUE THREAD

**Issue by Kyle-Ye** [firebase/firebase-ios-sdk#13199]
### Description

Recently, due to some invalid URL data passed from our server, our iOS projects received a lot of crash report on Firebase related library.

eg.
```
Fatal Exception: NSRangeException
*** -[__NSCFString substringToIndex:]: Index 9223372036854775807 out of bounds; string length 2269
```

> Note: 9223372036854775807 = 0x7FFFFFFFFFFFFFFF = NSNotFound

<img width="586" alt="image" src="https://github.com/firebase/firebase-ios-sdk/assets/43724855/03cfbdaa-645e-4e71-880f-f9bfe9f58af8">

### Reproducing the issue

Not yet.

### Firebase SDK Version

10.25.0

### Xcode Version

15.3

### Installation Method

Swift Package Manager

### Firebase Product(s)

Performance

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

    {
      "identity" : "firebase-ios-sdk",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/firebase-ios-sdk.git",
      "state" : {
        "revision" : "97940381e57703c07f31a8058d8f39ec53b7c272",
        "version" : "10.25.0"
      }
    },
```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by Kyle-Ye** [comment]
Hi, forgive me for not knowing much about the release pattern and rules of Firebase-iOS-SDK.

Do we have any plans or timeline for releasing a new version containing this issue's fix patch (eg. 10.28.2) @paulb777

**Comment by paulb777** [comment]
@Kyle-Ye The fix will release in 10.29.0, planned for the week of July 8. thanks again for the issue and fix!
